core/src/main/java/com/linecorp/armeria/internal/common/BuiltInDependency.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal.common;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface BuiltInDependency {
```

Comment:
How about renaming this to `CreateIfMissing` as the original issue suggested and applying it to user-specified classes as well?
A user might want to create a dependency using this annotation without DependencyInjector:
```java
@CreateIfMissing
public class MyRequestConverter implements ... {
    public MyRequestConverter() {} // Must have the default constructor that doesn't have any parameter.
}

public class MyAnnotatedService {
    @Get("/")
    @RequestConveter(MyRequestConverter.class)
    public HttpResponse foo(...) {...}
}
```

Reply:
I've changed the annotation name as per your comment and relocated it under a public package(`com.linecorp.armeria.server.annotation`).

But the current behavior of this annotation is different from what its name(`CreateIfMissing`) indicates because `BuiltInDependencyInjector` is applied first before any other `DependencyInjector` get a chance to resolve the dependencies.

Maybe we should put `BuiltInDependencyInjector` to the last?

Reply:
> Maybe we should put BuiltInDependencyInjector to the last?

I believe this is an essential change. We need to defer `NoSuchBeanDefinitionException` until `BuiltInDependencyInjector` fails to create the object.

Reply:
Yeah, we definitely need to put the `BuiltInDependencyInjector` at last. 
The injection should work even when a user specifies a customizing injector, we need to add delegating logic in `OrElseDependencyInjector`.
```java
public <T> T getInstance(Class<T> type) {
    try {
        final T instance = first.getInstance(type);
        if (instance != null) {
            return instance;
        }
    } catch (Exception e) {
        if (!hasCreateIfMissing(type)) {
            throw e;
        }
        logger.trace("...");
    }
    return second.getInstance(type);
}
```